src/libs/masks.c

```diff
@@ -248,11 +248,11 @@ static void _tree_group(GtkButton *button, dt_lib_module_t *self)
   GtkTreeModel *model = gtk_tree_view_get_model(GTK_TREE_VIEW(lm->treeview));
   GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(lm->treeview));
 
-  GList *items = g_list_first(gtk_tree_selection_get_selected_rows(selection, NULL));
   int pos = 0;
-  while(items)
+  GList *items = gtk_tree_selection_get_selected_rows(selection, NULL);
```

Comment:
These (_tree_group, _tree_intersection etc) look like good candidates for some deduplication and refactoring. This loop could then be switched to `gtk_tree_selection_selected_foreach`. I could put that on my todo list if it feels too Gtk-y for you?

Reply:
I know not quite enough about Gtk to be dangerous :-)
(yet).